src/sentry/incidents/endpoints/serializers/workflow_engine_incident.py

```diff
@@ -52,49 +97,136 @@ def get_incident_status(self, priority: int | None) -> int:
             raise ValueError("Priority is required to get an incident status")
         return self.priority_to_incident_status[priority]
 
-    def get_incident_activities(
+    def get_open_period_activities(
         self, open_period: GroupOpenPeriod
     ) -> list[IncidentActivitySerializerResponse]:
-        # TODO: Implement this
-        return [
-            {
+        # a metric issue will only have one openperiod because if it's reopened it'll make a new metric issue
```

Comment:
Metric issues can (and should) have multiple open periods - the OK -> HIGH -> MED -> OK cycle is one open period, but the second time it goes OK->HIGH->OK it'll create a new open period on the same issue. We're regressing/resolving the same issue each time.

I think each incident should be 1:1 with an open period?

Reply:
I think the only case where we'd create a new issue is if the underlying detector changes